core/dbt/config/runtime.py

```diff
@@ -314,10 +314,11 @@ def warn_for_unused_resource_config_paths(
             return
 
         msg = UNUSED_RESOURCE_CONFIGURATION_PATH_MESSAGE.format(
-            len(unused),
-            '\n'.join('- {}'.format('.'.join(u)) for u in unused)
+            printer.red(len(unused)),
```

Comment:
This will have to be `printer.red(str(len(unused)))` to satisify mypy, because `printer.red()` is annotated as taking a `str`.

You could alternatively update the type annotations in `printer.py` to note that `Any` is allowed as the `text` argument to `color`, `red`, `yellow`, and `green`.

Reply:
fwiw I think we should use `yellow` here -- yellow indicates a warning, red indicates an error

Reply:
I ended up taking this out as it doesn't really add to the task at hand and to make this PR more simple =)